src/v1.js

```diff
@@ -586,6 +587,59 @@ builder.declare({
   res.reply(roles.map(r => roleToJson(r, this)));
 });
 
+/** List role Ids **/
+builder.declare({
+  method:     'get',
+  route:      '/roleids/',
+  query: {
+    continuationToken: /./,
+    limit: /^[0-9]+$/,
+  },
+  name:       'listRoleIds',
+  input:      undefined,
+  output:     'list-role-ids-response.yml',
+  stability:  'stable',
+  title:      'List Role IDs',
+  description: [
+    'If no limit is given, the roleIds of all roles are returned. Since this',
+    'list may become long, callers can use the `limit` and `continuationToken`',
+    'query arguments to page through the responses.',
+  ].join('\n'),
+}, async function(req, res) {
+  let hashids = new Hashids();
+  let continuationToken = parseInt(req.query.continuationToken, 10);
+  let limit = parseInt(req.query.limit, 10);
+  let response = {};
+
+  // Load all roles
+  let roles = await this.Roles.get();
+  let length = roles.length;
+
+  // Slice the list of roles based on continuationToken and limit
+  if (continuationToken && limit) {
+    roles = roles.slice(continuationToken, limit+continuationToken);
+    continuationToken = limit+continuationToken;
+
+    if (continuationToken < length) {
+      response.continuationToken = hashids.encode(continuationToken);
```

Comment:
You can add more padding to the encoded value with the `encode` second parameter:

```js
response.continuationToken = hashids.encode(continuationToken, 10);
```

Reply:
Will do! I thought padding was not required so didn't add it initially.